site/source/tutorials/internationalization.md

```diff
@@ -0,0 +1,504 @@
+
+* `locale`: the default locale for the application
+* `supportedLocales`: an optional array of additional locales to include in the build
+* `messageBundles`: an optional array of any message bundles to include in the build
+
+Currently, the build does not ensure the `locale` specified in the `.dojorc` is set as the application's root locale; it is only used for bundling CLDR data and locale-specific message translations. This will change in a future release, but in the meantime, the initial locale must be set manually in the application with `@dojo/i18n/i18n.switchLocale`. Further, CLDR data are currently parsed from calls to `@dojo/i18n/cldr/load`, but this too will change in a future release to instead inject CLDR data from a `.dojorc` setting.
+
+To demonstrate the steps required to set the root locale and bundle all CLDR data and supported message bundles, suppose for our application that the default locale is English, but Spanish and French are also supported. Further, suppose that the application utilizes a single message bundle, located at `src/nls/main.ts`.
+
+```
+// .dojorc
+{
+	"build-webpack": {
+		"locale": "en",
+		"supportedLocales": [ "es", "fr" ],
+		"messageBundles": [ "src/nls/main" ]
+	}
+}
+```
+
+With this configuration, the translations located at `src/nls/main.ts`, `src/nls/es/main.ts`, and `src/nls/fr/main.ts` will be included in the build. As mentioned above, applying the default locale is still a manual process:
+
+```typescript
+import loadCldrData from '@dojo/i18n/cldr/load';
+import i18n, { switchLocale } from '@dojo/i18n/i18n';
+
+// Eventually this will be handled automatically, but for now this is required
+const defaultLocale = 'en';
+const supportedLocales = [ 'en', 'fr', 'es' ];
+const rootLocale = i18n.locale.split('-')[0];
+const isLocaleSupported = supportedLocales.some(locale => locale === rootLocale);
+
+// If the user's environment's locale is not one of the supported locales,
+// then switch to the default locale.
+if (!isLocaleSupported) {
+	switchLocale(defaultLocale);
+}
+
+// The build will parse these URLs, load the data for each supported locale,
+// include the data in the build, and ensure they are registered via
+// `@dojo/i18n/cldr/load`.
+export default loadCldrData([
+	'./cldr/main/{locale}/numbers.json',
+	'./cldr/main/{locale}/currencies.json',
+	'./cldr/supplemental/likelySubtags.json',
+	'./cldr/supplemental/plurals.json'
+]);
+```
+
+## Conclusion
+
+Internationalizing an application is not a simple task, and requires knowledge that not every developer has. While application developers must take the first step, Dojo 2 removes many of the barriers to creating a properly internationalized application, and therefore the uncertainty that an application may fail some of its users.
```

Comment:
In general we like to end with a way to download the code snippets, though that may not be practical here. Let's discuss with @vansimke and see if there's anything we should offer.